### PR TITLE
Add octet-stream (PEM and DER) and x-pkcs7-certificates to unpack_cert_content

### PR DESCRIPTION
**Pull request by fredericoschardong**
This PR addresses [the issues previsouly discussed](https://github.com/MatthiasValvekens/pyHanko/discussions/196#discussion-4648474) about nonstandard URL references in the `Authority Information Access` field of x509 certificates.

PS: I found a TLS certificate using `x-pkcs7-certificates` so I decided to add it as well, despite being deprecated.

### CODE REVIEW

File: tests/test_certificate_validator.py [hunk 1]
```diff
@@ -150,3 +153,20 @@ async def test_self_signed_with_policy(self):
         self.assertEqual(qp.user_domain_policy_id, 'any_policy')
         self.assertEqual(qp.issuer_domain_policy_id, 'any_policy')
 
+    async def test_basic_certificate_validator_tls_aia(self):
```
Comment (MatthiasValvekens) [inline tests/test_certificate_validator.py]: Any chance you could download these end-entity certs and commit them as part of the test suite? Apparently there's a bug in the macOS-specific part of the TLS handling in `oscrypto` causing the handshake to fail.

I'd also prefer only testing the fetcher here (and moving the test to the appropriate place for that). Given that this test depends on external data that we don't control, I want to keep the number of moving parts to the absolute minimum here :).

File: pyhanko_certvalidator/fetchers/common_utils.py [hunk 2]
```diff
@@ -27,19 +27,36 @@
...
 
 ACCEPTABLE_CERT_PEM_ALIASES = (
-    'application/x-pem-file', 'text/plain',
+    'application/x-pem-file', 'text/plain', 'application/octet-stream'
 )
 
 
 def unpack_cert_content(response_data: bytes, content_type: Optional[str],
                         url: str, permit_pem: bool):
+    def unpack_der_pkcs7(pkcs7_data: bytes, pkcs7_url: str):
```
Comment (MatthiasValvekens) [inline pyhanko_certvalidator/fetchers/common_utils.py]: Minor nit: maybe make this a top-level function with an underscore name prefix? I don't like using nested functions unless lexical closures are somehow necessary.

**Review by MatthiasValvekens — CHANGES_REQUESTED** [review]
Good work, thanks! The test still needs some attention, though. I don't have time to fix it myself right this instant, but I wouldn't mind fixing the remaining issues later if you don't get to them by then :).

**Comment by fredericoschardong** [comment]
> Good work, thanks! The test still needs some attention, though. I don't have time to fix it myself right this instant, but I wouldn't mind fixing the remaining issues later if you don't get to them by then :).

Done. Please let me know if there is anything further I can do to enhance.

**Review by MatthiasValvekens — APPROVED** [review]

**Comment by MatthiasValvekens** [comment]
Hi @fredericoschardong,


Thanks! I made a couple further changes:

 - Moved the test to `test_registry.py` and only test path building instead of full validation (we're working with a bunch of "real" certs with relatively short lifetimes, so this makes the test easier to maintain)
 - Changed the order of the branches so that the PKCS#7 DER case is evaluated first. This didn't matter previously because there was no overlap between the content type groups, but since `application/octet-stream` is in several groups, we have to be a bit more careful.
 - General cleanup to make the code easier to merge.

If you're OK with these changes I'll squash them and merge them into the `master` branch, while also keeping them around for a potential 0.19.7 release.

**Comment by fredericoschardong** [comment]
Thank you, @MatthiasValvekens. 

Because we are no longer bound by SSL certificates, I added the certificate that triggered this PR. In addition, I converted the certificates to DER to make the loading procedures easier.

Please let me know if you want anything changed. Everything appears to be in order to me.

**Comment by MatthiasValvekens** [comment]
I squashed your commit and merged it into `master` with modifications for the changes to the API & test setup over there. Thank you!

I'll probably do at least one more patch release on the 0.19.x branch before releasing the work that's currently on `master`, hence why I asked to branch off this earlier tag. Is this issue currently blocking you?

EDIT: I figured I might as well release `0.19.7` right now. I've also updated the dependency in pyHanko, but haven't released that one yet. Can be arranged if it's a blocker for you, though. I just about have enough material for a (small) minor version bump over there.

**Comment by fredericoschardong** [comment]
> I'll probably do at least one more patch release on the 0.19.x branch before releasing the work that's currently on `master`, hence why I asked to branch off this earlier tag. Is this issue currently blocking you?
> 
> EDIT: I figured I might as well release `0.19.7` right now. I've also updated the dependency in pyHanko, but haven't released that one yet. Can be arranged if it's a blocker for you, though. I just about have enough material for a (small) minor version bump over there.

Yes, it is blocking me. I would greatly appreciate it if you could make a new release (of pyHanko) that incorporates our changes.

**Pull request merged**